src/main/kotlin/eu/europa/ec/eudi/pidissuer/adapter/out/jose/EncryptCredentialResponseWithNimbus.kt

```diff
@@ -45,16 +40,15 @@ class EncryptCredentialResponseWithNimbus(
 ) : EncryptCredentialResponse {
 
     override fun invoke(
-        response: IssueCredentialResponse.PlainTO,
         parameters: RequestedResponseEncryption.Required,
-    ): Result<IssueCredentialResponse.EncryptedJwtIssued> = runCatching {
+        responseAsJwtClaims: JWTClaimsSet.Builder.() -> Unit,
```

Comment:
This transfers the responsibility of assembling the JWT to the caller. Also leaks the implementation